### ISSUE TITLE
Expand deletion UI/functionality

### ISSUE THREAD

**Issue by pdehaan** [mozilla/activity-stream#105]
Maybe I'm using a shared family computer and looking at gifts (or whatever) and don't want those results to appear on my homepage where other people could see them.

Not sure how to remove individual items from this homepage without clearing ALL my browsing history. It'd be nice if when you hovered over an item there was a floating little X which would allow you to delete said item and replace it with the next most recent item (so there are always 6 Top Sites listed, or 3 Spotlight results or whatever.

**Comment by nchapman** [comment]
I just talked about this with @bryanbell. He's working on the UX.

**Comment by pdehaan** [comment]
Dear @nchapman,
![](http://2ksjbh44hk54113rf2pk9bis06.wpengine.netdna-cdn.com/wp-content/uploads/2014/04/747958.gif)

Sincerely,
@pdehaan

**Comment by bryanbell** [comment]
For now, let's just do this.  We'll have to augment it in the near future when we add a contextual menu.

Video example: 
http://cl.ly/1x1Q0Q2T0o38

Code Example: 
[activity-stream.zip](https://github.com/mozilla/activity-streams/files/136783/activity-stream.zip)

**Comment by oyiptong** [comment]
Should the blocking be by URL or by site?

The semantics are important. e.g. Do I not want to see any pages by BuzzFeed, or just this page?
Top Sites blocking should block the site or the link?

While Top Sites shows a site, we show the most frecent link for it.

**Comment by bryanbell** [comment]
For this version we're only going to bother supporting single item delete.

**Comment by k88hudson** [comment]
Implementation here is the UI with a message that pops up saying "We're working on deletion!"

Pushing real deletion until a future milestone

**Comment by k88hudson** [comment]
@bryanbell can you add UI for spotlight as well

**Comment by bryanbell** [comment]
<img width="287" alt="pasted_image_2_19_16__11_31_am" src="https://cloud.githubusercontent.com/assets/228979/13181704/53420a46-d6fc-11e5-8c34-4a24f8f376e4.png">

Delete for Spotlight items

**Comment by bryanbell** [comment]
delete mock http://people.mozilla.org/~shorlander/activity-stream/activity-stream-04/start-page.html#

**Comment by oyiptong** [comment]
@bryanbell will we also have an "unbookmark" functionality?

**Comment by oyiptong** [comment]
(we if we don't, great ;-) 🎣)

**Comment by k88hudson** [comment]
@oyiptong @bryanbell I added a ticket for bookmarking/unbookmarking to the post-v1 miletstone: https://github.com/mozilla/activity-streams/issues/162

**Comment by oyiptong** [comment]
:+1:

**Comment by k88hudson** [comment]
Let's wait 'til https://github.com/mozilla/activity-streams/issues/163 has landed for this:

Also, there's going to be real delete api once this lands: https://github.com/mozilla/activity-streams/pull/166/files

**Comment by pdehaan** [comment]
Random question, but I noticed in [shorlander's mockup](http://people.mozilla.org/~shorlander/activity-stream/activity-stream-04/start-page.html#) that we have two options for deleting items from the activity Stream:

1. **Remove from Activity Stream**
2. **Block Site from Activity Stream**

If I [accidentally] block a site from my activity stream, where are we going to put the UI to undelete the blocked site? Or would I just never see YouTube.com links ever again?

I don't expect to be able to recover a deleted history item, but if Activity Streams is managing some sort of blocklist of sites that I manually blocked, I expect to be able to undo that. (Much like how I can "undo" push notification blocklists in `about:preferences#content`).

![firefox_start](https://cloud.githubusercontent.com/assets/557895/13274160/3ba5f842-da5e-11e5-925a-5b9a76523304.png)

**Comment by oyiptong** [comment]
You mean, how do we put a UI to unblock the site? I don't know. I'm not a UI designer and all I can think of is yet another piece of management UI. (Thankfully, I don't design UIs ;-))

The current newtab page does a hatchet job: it only allows you to unblock all sites. It sucks, but it does its job.

**Comment by k88hudson** [comment]
I filed a new ticket with specific requirements for MVP here: https://github.com/mozilla/activity-streams/issues/221

Let's push this post-MVP to expand via context menus, etc.

**Comment by k88hudson** [comment]
I'm closing this since we've landed #701 and are tracking new work in #778
